hack/verify-generated-json-codecs.sh

```diff
@@ -1,9 +1,15 @@
 #!/bin/bash
 source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
-echo "===== Verifying Generated Ugorji JSON Codecs ====="
+function cleanup() {
+    return_code=$?
+    os::test::junit::generate_report
+    os::util::describe_return_code "${return_code}"
+    exit "${return_code}"
+}
+trap "cleanup" EXIT
 
-output=$(find ${OS_ROOT}/vendor/k8s.io/kubernetes -name "types.generated.go")
-if [[ -n "${output}" ]]; then
-  os::log::fatal "FAILURE: Verification of existing ugorji JSON codecs failed. These should NOT exist:\n${output}"
-fi
+os::test::junit::declare_suite_start "verify/codecs"
+# no ugorji codecs should be checked in
+os::cmd::expect_success_and_not_text "find ${OS_ROOT}/vendor/k8s.io/kubernetes -name 'types.generated.go'" "."
```

Comment:
is there a way to keep the error strings?

Reply:
Output of `os::cmd` shows `std{out,err}` if something goes wrong -- do you mean in the success case?

Reply:
Or, did you mean the verbose failure? Not really a great way to keep it other than to:

```sh
if [[ "${return_code}" -ne 0 ]]; then
    os::log::warning "whatever"
fi
```

In the `cleanup()` trap, but in general this was the only file where the error was not "this failed, run the update"